geomstats/information_geometry/multivariate_normal.py

```diff
@@ -225,6 +351,125 @@ def pdf(x):
+
+    def sample(self, point, n_samples=1):
+        """Sample from a multivariate normal distribution.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., n + n**2]
+            Point representing a multivariate normal distribution. First $n$
+            elements contain the mean vector and the $n**2$ last elements contain
+            the covariance matrix row by row.
+        n_samples : int
+            Number of points to sample with each parameter in point.
+            Optional, default: 1.
+
+        Returns
+        -------
+        samples : array-like, shape=[..., n_samples, n]
+            Sample from multivariate normal distributions.
+        """
+        # geomstats.errors.check_belongs(point, self)
+        locs, covs = self.reformat(point)
+        locs = gs.to_ndarray(locs, to_ndim=2)
+        covs = gs.to_ndarray(covs, to_ndim=3)
+        samples = []
+        for loc, cov in zip(locs, covs):
+            samples.append(gs.array(multivariate_normal.rvs(loc, cov, size=n_samples)))
+        samples = samples[0] if point.shape[0] == 1 else gs.stack(samples)
+        return gs.squeeze(samples)
+
+    def point_to_pdf(self, point):
+        """Compute pdf associated to point.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., n + n**2]
+            Point representing a multivariate normal distribution. First $n$
+            elements contain the mean vector and the $n**2$ last elements contain
+            the covariance matrix row by row.
+
+        Returns
+        -------
+        pdf : function
+            Probability density function of the multivariate normal
+            distributions with parameters provided by point.
+        """
+        # geomstats.errors.check_belongs(point, self)
+        if point.ndim > 3:
+            raise NotImplementedError
+        n = self.n
+
+        def pdf(x):
+            """Generate parameterized function for normal pdf.
+
+            Parameters
+            ----------
+            x : array-like, shape=[n_samples, n]
+                Points at which to compute the probability
+                density function.
+            """
+            x = gs.to_ndarray(x, to_ndim=2, axis=0)
+            loc, cov = self.reformat(point)
+            loc = gs.to_ndarray(loc, to_ndim=2)
+            cov = gs.to_ndarray(cov, to_ndim=3)
+            det_cov = gs.linalg.det(cov)
+            inv_cov = gs.linalg.inv(cov)
+            pdf_normalization = 1 / gs.sqrt(gs.power(2 * gs.pi, n) * det_cov)
+            pdf = []
+            for xi in x:
+                xi0 = xi - loc
+                pdf_at_xi = gs.exp(-0.5 * xi0[:, None, :] @ inv_cov @ xi0[:, :, None])
+                pdf.append(gs.squeeze(pdf_at_xi))
+            pdf = gs.stack(pdf)
+            pdf = pdf_normalization * pdf
+            return gs.squeeze(pdf)
+
+        return pdf
+
+
+class MultivariateCenteredNormalMetric(NFoldMetric):
```

Comment:
Maybe not for this PR, but for readability we should create:

`class ScaleMetric(NFoldMetric)` with `n_copies=1` and `scale`. Otherwise it's too much of a "hack" for the user to have to think about creating a nfold manifold with one copy. Maybe raise an issue on this?

@lpereira95

Reply:
Why doesn't this inherit from FisherRaoMetric?

Reply:
> Why doesn't this inherit from FisherRaoMetric?

@ninamiolane It seems that as it is, FisherRaoMetric is only defined for one-dimensional sample spaces ? It would be nice to extend it to multi-dimensional sample spaces, and use it in the tests.